packages/service-library/src/servicelib/rabbitmq.py

```diff
@@ -152,3 +186,66 @@ async def publish(self, exchange_name: str, message: Message) -> None:
                 aio_pika.Message(message.encode()),
                 routing_key="",
             )
+
+    async def rpc_request(
+        self,
+        namespace: RPCNamespace,
+        method_name: str,
+        *,
+        timeout: Optional[PositiveInt] = 5,
+        **kwargs: dict[str, Any],
+    ) -> Any:
+        """
+        Call a remote registered `handler` by providing it's `namespace`, `method_name`
+        and list of expected arguments.
+
+        NOTE: `namespace` should always be composed via `get_namespace`
+        """
+
+        if not self._rpc:
+            raise RPCNotInitializedError()
+
+        namespaced_method_name = _get_namespaced_method_name(namespace, method_name)
+        try:
+            queue_expiration_timeout = timeout
+            awaitable = self._rpc.call(
+                namespaced_method_name,
+                expiration=queue_expiration_timeout,
+                kwargs=kwargs,
+            )
+            return await asyncio.wait_for(awaitable, timeout=timeout)
+        except MessageProcessError as e:
+            if e.args[0] == "Message has been returned":
+                raise RemoteMethodNotRegisteredError(
+                    method_name=namespaced_method_name, incoming_message=e.args[1]
+                ) from e
+            raise e
+
+    async def rpc_register(
+        self, namespace: RPCNamespace, method_name: str, handler: Awaitable
+    ) -> None:
+        """
+        Bind a local `handler` to a `namespace` and `method_name`.
+        The handler can be remotely called by providing the `namespace` and `method_name`
+
+        NOTE: method_name could be computed from the handler, but by design, it is
+        left to the caller to do so.
+        NOTE: `namespace` should always be composed via `get_namespace`
+        """
```

Comment:
following these two notes, if you really do not want to change this interface, why not add an extra helper that enforce those?
```python
async def rpc_register_foo ( self, entries: dict,  handler: Awaitable):
     await self.rpc_register(get_namespace(entries), method_name= handler.__name__, handler=handler)
     
```

Reply:
I like this suggestion. Let's go for this